src/code/z_actor.c

```diff
@@ -457,439 +1269,3558 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BD9A0.s")
+void func_800B8E58(Player* player, u16 sfxId) {
+    if (player->currentMask == PLAYER_MASK_GIANT) {
+        func_8019F170(&player->actor.projectedPos, sfxId);
+    } else {
+        Audio_PlaySfxGeneral(sfxId, &player->actor.projectedPos, 4, &D_801DB4B0, &D_801DB4B0, &D_801DB4B8);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BD9E0.s")
+void Actor_PlaySfxAtPos(Actor* actor, u16 sfxId) {
+    func_8019F1C0(&actor->projectedPos, sfxId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDAA0.s")
+void func_800B8EF4(GlobalContext* globalCtx, Actor* actor) {
+    u32 sfxId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDB6C.s")
+    if (actor->bgCheckFlags & 0x20) {
+        if (actor->depthInWater < 20.0f) {
+            sfxId = NA_SE_PL_WALK_WATER0 - SFX_FLAG;
+        } else {
+            sfxId = NA_SE_PL_WALK_WATER1 - SFX_FLAG;
+        }
+    } else {
+        sfxId = func_800C9BDC(&globalCtx->colCtx, actor->floorPoly, actor->floorBgId);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDC5C.s")
+    func_8019F1C0(&actor->projectedPos, NA_SE_EV_BOMB_BOUND);
+    func_8019F1C0(&actor->projectedPos, sfxId + SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDCF4.s")
+void func_800B8F98(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x02;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/Actor_Noop.s")
+void func_800B8FC0(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 4;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BDFC0.s")
+void func_800B8FE8(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x08;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800BE03C.s")
+void func_800B9010(Actor* actor, u16 sfxId) {
+    actor->sfxId = sfxId;
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x01;
+}
 
-Actor* func_800BE0B8(GlobalContext* globalCtx, Actor* inActor, s16 arg2, u8 arg3, f32 arg4) {
-    Actor* actor = globalCtx->actorCtx.actorList[arg3].first;
+void func_800B9038(Actor* actor, s32 timer) {
+    actor->audioFlags &= ~(0x10 | 0x08 | 0x04 | 0x02 | 0x01);
+    actor->audioFlags |= 0x10;
+
+    // The sfxId here are not actually sound effects, but instead this is data that gets sent into
+    // the io ports of the music macro language (func_801A0810 / Audio_PlaySfxAtPosWithSoundScriptIO is
+    // the function that it's used for)
+    if (timer < 40) {
+        actor->sfxId = 3;
+    } else if (timer < 100) {
+        actor->sfxId = 2;
+    } else {
+        actor->sfxId = 1;
+    }
+}
```

Comment:
```suggestion
    // The sfx here are a special case that are not actually sound effect Ids.
    // Instead, this is an index for a lookup-table in audio sequence 0 that 
    // stores the note duration for the timer sfx. These indices are accessed 
    // by audio sequence 0 though an io port (port 6) set by func_801A0810
    // and are interpreted by the music macro language
    if (timer < 40) {
        actor->sfx = 3; // timer index = 2 (note duration index)
    } else if (timer < 100) {
        actor->sfx = 2; // timer index = 1 (note duration index)
    } else {
        actor->sfx = 1; // timer index = 0 (note duration index)
    }
}
```
https://discord.com/channels/688807550715560050/688851451014938642/911268505452707851

I brought this up in OoT since it also applies to them, and the only suggestion recommended was to leave it named `sfx` instead of `sfxId`. I think that's fine for now and this can be addressed later in later doc passes. 

This is my attempt at documenting this exception. I would change the comment in the header file to:
`/* 0x050 */ u16 sfx; // Either a sound effects Id or in the special case of (audioFlags & 0x10): this represents an note-duration index for the timer sfx`